app/javascript/spec/auth-key-pair-cloud/auth-key-pair-cloud-form.spec.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import toJson from 'enzyme-to-json';
+import fetchMock from 'fetch-mock';
+import { shallow } from 'enzyme';
+
+import { act } from 'react-dom/test-utils';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import '../helpers/mockAsyncRequest';
+import '../helpers/miqSparkle';
+import '../helpers/miqAjaxButton';
+import '../helpers/miqFlashLater';
+import { mount } from '../helpers/mountForm';
+import AuthKeypairCloudForm from '../../components/auth-key-pair-cloud/index';
+
+
+describe('Add testcases for creating new auth key pair', () => {
+  let submitSpyMiqSparkleOn;
+  let submitSpyMiqSparkleOff;
+  const emsList = {
+    ems_choices: [
+      { name: 'name1', id: 1 },
+      { name: 'name2', id: 2 },
+    ],
+  };
+
+  const values = {
+    name: 'key1',
+    public_key: 'abc',
+    resource_id: 2,
+  };
+  beforeEach(() => {
+    submitSpyMiqSparkleOn = jest.spyOn(window, 'miqSparkleOn');
+    submitSpyMiqSparkleOff = jest.spyOn(window, 'miqSparkleOff');
+  });
+
+  afterEach(() => {
+    fetchMock.reset();
+    fetchMock.restore();
+    submitSpyMiqSparkleOn.mockRestore();
+    submitSpyMiqSparkleOff.mockRestore();
+  });
+
+  it('should render a auth key pair form', (done) => {
+    fetchMock
+      .mock('/auth_key_pair_cloud/ems_form_choices', emsList);
+    const wrapper = shallow(<AuthKeypairCloudForm />);
+    setImmediate(() => {
+      wrapper.update();
+      expect(toJson(wrapper)).toMatchSnapshot();
+      done();
+    });
+  });
+
+  it('should correctly add new key pair .', async(done) => {
+    fetchMock
+      .mock('/auth_key_pair_cloud/ems_form_choices', emsList);
+    fetchMock.postOnce('/api/auth_key_pairs', values);
+    const wrapper = mount(<AuthKeypairCloudForm />);
+    wrapper.find('input').at(0).simulate('change', { target: { value: 'key1' } });
+    wrapper.update();
+
+    setTimeout(async() => {
```

Comment:
Please avoid using `setTimeout`, I'm pretty sure this can be done without it if you're already using `await act`...